docs/library/graphing/plotly.md

```diff
@@ -100,14 +44,82 @@ z_data = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/a
 fig = go.Figure(data=[go.Surface(z=z_data.values)])
 fig.update_traces(contours_z=dict(show=True, usecolormap=True,
                                   highlightcolor="limegreen", project_z=True))
-fig.update_layout(scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
-                  width=500, height=500,
-                  margin=dict(l=65, r=50, b=65, t=90)
+fig.update_layout(
+    scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
+    margin=dict(l=65, r=50, b=65, t=90)
 )
+
+def mountain_surface():
+    return rx.center(
+        rx.plotly(data=fig),
+    )
+```
+
+## Plot as State Var
+
+The figure need to be set as a state var if it needs to be updated during run time. 
+
+```python demo exec
+import plotly.express as px
+class PlotlyState(rx.State):
+    df = px.data.gapminder().query(f"country=='Canada'")
+    figure = px.line(
+        df,
+        x="year",
+        y="lifeExp",
+        title="Life expectancy in Canada",
+    )
+
+    def set_selected_country(self, country):
+        self.df = px.data.gapminder().query(f"country=='{country}'")
+        self.figure = px.line(
+            self.df,
+            x="year",
+            y="lifeExp",
+            title=f"Life expectancy in {country}",
+        )
+
+
+def line_chart_with_state():
+    return rx.vstack(
+        rx.select(
+            ['China', 'France', 'United Kingdom', 'United States', 'Canada'],
+            default_value="United States",
```

Comment:
```suggestion
            default_value="Canada",
```

The figure is initialized with Canada data, so the `default_value` for the select should match the initial state of the graph (even though it won't stick if the page is refreshed)